src/Simulation/QSharpFoundation/Diagnostics/Assert.qs

```diff
@@ -25,7 +25,15 @@ namespace Microsoft.Quantum.Diagnostics {
     /// - Microsoft.Quantum.Diagnostics.AssertMeasurementProbability
     operation AssertMeasurement(bases : Pauli[], qubits : Qubit[], result : Result, msg : String) : Unit
     is Adj + Ctl {
-        body intrinsic;
+        body (...) {
+            AssertMeasurementProbability(bases, qubits, result, 1.0, msg, 1e-10);
+        }
+        adjoint (...) {
+            // Empty.
+        }
```

Comment:
```suggestion
        adjoint self;
```
For why this is the case, the condition being checked by `AssertMeasurement` should be the same whether we got there by starting at the end of an operation and running backwards or starting at the beginning and running forwards. In both cases, we got to the same point in time, and thus the state of the register should be the same.

Reply:
I assume that your proposed change does not contradict the [fragment](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.diagnostics.assertmeasurement#remarks) "_Note that the Adjoint and Controlled versions of this operation will not check the condition_".

Reply:
Fixed.